proxy/coco/src/peer/subroutines.rs

```diff
@@ -349,28 +354,36 @@ async fn query(urn: RadUrn, state: State, mut sender: mpsc::Sender<Input>) {
 }
 
 /// Run a clone for the given `url`. On completion report back with the success or failure.
-async fn clone(url: RadUrl, state: State, mut sender: mpsc::Sender<Input>) {
+async fn clone(urn: Urn, remote_peer: PeerId, state: State, mut sender: mpsc::Sender<Input>) {
     sender
-        .send(Input::Request(input::Request::Cloning(url.clone())))
+        .send(Input::Request(input::Request::Cloning(
+            urn.clone(),
+            remote_peer,
+        )))
         .await
         .ok();
 
-    match state.clone_project(url.clone(), None).await {
+    match state
+        .clone_project(urn.clone(), remote_peer, None, None)
```

Comment:
@xla: I opted for defaulting the `fetch::Limit` -- introduced here https://github.com/radicle-dev/radicle-upstream/pull/1463/commits/ddb6dae3d973151a07061bcabcf8cffde7fcca22 -- but it would be nice to be able to change this at runtime as well.

Any thoughts on how this could be injected into the subroutines at runtime? Another channel perhaps?

Reply:
Could you expand on what potential scenarios look like that would constitute runtime configuration changes? The reason I'm asking is that neither this comment or the original commit message reason about the why such a capability is desried.

> Any thoughts on how this could be injected into the subroutines at runtime? Another channel perhaps?

Yeah usually configuration changes over time would come through some form of channel. For the configured seeds we use a `watch` channel to provide last value semantics. Would need to look deeper into this use-case to give proper recommendation. There might be a general need for a latest configuration made to be known to the internals of the peer.

Reply:
So a use case would be the following scenario.

We ask to clone the Linux project, but as we're doing so the fetcher throws an error and says we've exceeded the configured limits. We're sure that this really is the Linux project and not a bunch of spammy data, so we configure our limits to go higher, try the replication again, and once we've got the project we can reset the limit.

Does that help shed some light on the question here?

Reply:
- [ ] Somewhat. Will the limit go down again after the linux kernel has been replicated from `pc-torvalds01`. By the sound of it this has to do with trust and spam protection. Which seems odd to be bound to runtime configuration changes. As in where would the signal come from that the fetch limit can be increased. I don't want to challenge the scenario too specifically, just curious to see how the concerns align

Reply:
> Somewhat. Will the limit go down again after the linux kernel has been replicated from pc-torvalds01.

Ya you would want to turn it down again.

> By the sound of it this has to do with trust and spam protection.

Exactly, it's exactly a trust mechanism.

> Which seems odd to be bound to runtime configuration changes. As in where would the signal come from that the fetch limit can be increased.

As @kim and I deliberated on it, we landed on "sane" default values and should be configurable depending on the scenario -- giving the malkovich the final say when using `replicate`.

How are you thinking about it?